backend/celery_worker.py

```diff
@@ -180,38 +179,113 @@ def ping_telemetry():
     maybe_send_telemetry("ping", {"ping": "pong"})
 
 
-@celery.task
-def process_integration_brain_sync():
-    integration = IntegrationBrain()
-    integrations = integration.get_integration_brain_by_type_integration("notion")
-
-    time = datetime.now(timezone.utc)  # Make `time` timezone-aware
-    # last_synced is a string that represents a timestampz in the database
-    # only call process_integration_brain_sync_user_brain if more than 1 day has passed since the last sync
-    if not integrations:
-        return
-    # TODO fix this
-    # for integration in integrations:
-    #     print(f"last_synced: {integration.last_synced}")
-    #     print(f"Integration Name: {integration.name}")
-    #     last_synced = datetime.strptime(
-    #         integration.last_synced, "%Y-%m-%dT%H:%M:%S.%f%z"
-    #     )
-    #     if last_synced < time - timedelta(hours=12) and integration.name == "notion":
-    #         process_integration_brain_sync_user_brain.delay(
-    #             brain_id=integration.brain_id, user_id=integration.user_id
-    #         )
+@celery.task(name="check_if_is_premium_user")
+def check_if_is_premium_user():
+    supabase = get_supabase_db()
+    supabase_db = supabase.db
+    # Get the list of subscription active
+    subscriptions = (
+        supabase_db.table("subscriptions")
+        .select("*")
+        .filter(
+            "current_period_end",
+            "gt",
+            datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f"),
+        )
+        .execute()
+    ).data
+    logger.debug(f"Subscriptions: {subscriptions}")
+
+    # Get List of all customers
+    customers = (
+        supabase_db.table("customers")
+        .select("*")
+        .order("created", desc=True)
+        .execute()
+        .data
+    )
+    unique_customers = {}
+    for customer in customers:
+        if customer["email"] not in unique_customers:
+            unique_customers[customer["email"]] = customer
+    customers = list(unique_customers.values())
+    logger.debug(f"Unique Customers with latest created date: {customers}")
+
+    # Matching Products
+    matching_product_settings = (
+        supabase_db.table("product_to_features").select("*").execute()
+    ).data
+    logger.debug(f"Matching product settings: {matching_product_settings}")
+
+    # if customer.id in subscriptions.customer then find the user id in the table users where email = customer.email and then update the user_settings with is_premium = True else delete the user_settings
+
+    for customer in customers:
+        logger.debug(f"Customer: {customer}")
+        # Find the subscription of the customer
+        user_id = None
+        matching_subscription = [
+            subscription
+            for subscription in subscriptions
+            if subscription["customer"] == customer["id"]
+        ]
+        logger.debug(f"Matching subscription: {matching_subscription}")
+        user_id = (
```

Comment:
Could be a join on the customer<->user on the email so we can get all at the same time. Not critical for now! maybe just a TODO 😄

Reply:
Not possible with supabase without a foreign key. Being foreign table I can't do that